skafka/src/main/scala/com/evolutiongaming/skafka/consumer/RebalanceCallback.scala

```diff
@@ -0,0 +1,333 @@
+package com.evolutiongaming.skafka.consumer
+
+import java.lang.{Long => LongJ}
+import java.time.Instant
+import java.util.{Map => MapJ}
+
+import cats.data.{NonEmptyMap => Nem, NonEmptySet => Nes}
+import cats.implicits._
+import cats.{StackSafeMonad, ~>}
+import com.evolutiongaming.catshelper.CatsHelper._
+import com.evolutiongaming.catshelper.ToTry
+import com.evolutiongaming.skafka.Converters._
+import com.evolutiongaming.skafka._
+import com.evolutiongaming.skafka.consumer.ConsumerConverters._
+import com.evolutiongaming.skafka.consumer.RebalanceCallback.Helpers._
+import com.evolutiongaming.skafka.consumer.RebalanceCallback.RebalanceCallbackOps
+import org.apache.kafka.clients.consumer.{Consumer => ConsumerJ, OffsetAndMetadata => OffsetAndMetadataJ}
+import org.apache.kafka.common.{TopicPartition => TopicPartitionJ}
+
+import scala.annotation.tailrec
+import scala.concurrent.duration.FiniteDuration
+import scala.jdk.CollectionConverters._
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Describes computations in callback methods of [[RebalanceListener1]].
+  *
+  * `RebalanceCallback` is just a data structure (a description of things to be done),
+  * so calling `RebalanceCallback.seek(...)` for example does not execute the `seek` right away.
+  *
+  * However all consumer related methods are executed on a `consumer.poll(...)` thread,
+  * at the time of interpretation/execution of the `RebalanceCallback` data structure.
+  *
+  * The computations' result is awaited on a `poll` thread,
+  * just as it would with blocking java API of `ConsumerRebalanceListener`.
+  *
+  * Errors from consumer related methods are thrown in a `poll` thread,
+  * and currently there's no way to provide recovering code for [[RebalanceCallback]].
```

Comment:
could you please add a link to an issue about adding `recovering` feature

Reply:
done in 9a5829e5